src/adapter/src/command.rs

```diff
@@ -278,18 +280,155 @@ impl IntoIterator for GetVariablesResponse {
     }
 }
 
-/// Type of closure that is called to validate a webhook request.
-pub type AppendWebhookValidation = Box<
-    dyn FnOnce(mz_repr::Datum, mz_repr::Datum) -> Result<bool, EvalError>
-        + Send
-        + std::panic::UnwindSafe,
->;
+/// Errors returns when running validation of a webhook request.
+#[derive(thiserror::Error, Debug)]
+pub enum AppendWebhookError {
+    // A secret that we need for validation has gone missing.
+    #[error("could not read a required secret")]
+    MissingSecret,
+    // Note: we should _NEVER_ add more detail to this error, including the actual error we got
+    // when running validation. This is because the error messages might contain info about the
+    // arguments provided to the validation expression, we could contains user SECRETs. So by
+    // including any more detail we might accidentally expose SECRETs.
+    #[error("validation failed")]
+    ValidationError,
+    // Note: we should _NEVER_ add more detail to this error, see above as to why.
+    #[error("validation panicked")]
+    ValidationPanicked,
```

Comment:
I folded `ValidationPanicked` into `InternalError`, I kept `InternalError` though so we can report 500s to the user instead of 400s in the case of `ValidationError`